pkg/controller/servicebinding/reconciler.go

```diff
@@ -157,16 +170,55 @@ func (r *reconciler) Reconcile(request reconcile.Request) (reconcile.Result, err
 		r.restMapper,
 	)
 	if err != nil {
+
+		_ = updateSBRConditions(r.dynClient, sbr,
+			conditionsv1.Condition{
+				Type:    CollectionReady,
+				Status:  corev1.ConditionFalse,
+				Reason:  FailedToGatherServiceInformationReason,
+				Message: err.Error(),
+			},
+			conditionsv1.Condition{
+				Type:    InjectionReady,
+				Status:  corev1.ConditionFalse,
+				Reason:  FailedToGatherServiceInformationReason,
+				Message: err.Error(),
+			},
+			conditionsv1.Condition{
+				Type:    BindingReady,
+				Status:  corev1.ConditionFalse,
+				Reason:  FailedToGatherServiceInformationReason,
+				Message: err.Error(),
+			},
```

Comment:
We did not even reach the state when we can say if something is injection ready or not, because we did not gather service data at all. Conditions must not exist at all. We should update them only if there is a change in status, and in that case we are going to update last transition time as well. In this PR, it looks like we have 3 conditions, but they all share value and reason. Why then not collapse it into single condition.

BTW, excellent read about conditions: https://dev.to/maelvls/what-the-heck-are-kubernetes-conditions-for-4je7